component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/util/Utils.java

```diff
@@ -246,4 +253,41 @@ public static EmailOtpServerException handleServerException(Constants.ErrorMessa
         }
         return new EmailOtpServerException(error.getMessage(), description, error.getCode());
     }
+
+    /**
+     * Check whether a given user is locked.
+     *
+     * @param user The user.
+     * @return True if user account is locked.
+     */
+    public static boolean isAccountLocked(User user) throws EmailOtpServerException {
+
+        try {
+            return EmailOtpServiceDataHolder.getInstance().getAccountLockService().isAccountLocked(user.getUsername(),
```

Comment:
is this null safe?

Reply:
yes it is null safe as the AccountLockService is expected to be always available while the component is active

Reply:
i actually meant the user object.. since this is a public method, I was wondering perhaps we could add a null check for the user object before calling its methods. wdyt?

Reply:
since this is a public method, added a null check